lib/Dialect/TritonGPU/IR/LinearLayoutConversions.cpp

```diff
@@ -1444,26 +1445,47 @@ LinearLayout getSM120DotScaledScaleLayout(
   const int totalWarps = mWarps * nWarps;
   const unsigned mRep_warp = tilesPerWarp[mIndex];
   const unsigned nRep_warp = tilesPerWarp[nIndex];
-  const unsigned kRep = std::min<unsigned>(kSize, 2);
-
+  // Each scale applies to K / groupSize consecutive elements along the K
+  // dimension. Examples:
+  // - FP8: groupSize = 1, K = 32 => 32 elements per scale
+  // - MXFP4 (int8 scales): groupSize = 2, K = 64 => 32 elements per scale
+  // - NVFP4 (float8 scales): groupSize = 4, K = 64 => 16 elements per scale
+  const unsigned kRep = kSize / groupSize;
   std::vector<std::vector<int32_t>> registerBase;
   std::vector<std::vector<int32_t>> laneBase;
   std::vector<std::vector<int32_t>> warpBase;
   if (dotOperandIdx == 0) { // per-row A-scale
     laneBase = {{0, 8}, {0, 0}, {0, 1}, {0, 2}, {0, 4}};
+    if (groupSize == 2)
+      registerBase.push_back({1, 0});
+    else if (groupSize == 4) {
+      registerBase.push_back({1, 0});
+      registerBase.push_back({2, 0});
+    }
     for (int offset = instrM * mWarps; offset < instrM * mWarps * mRep_warp;
          offset <<= 1)
       registerBase.push_back({0, offset});
+    for (int k = 1; k < kRep; k += 1)
```

Comment:
this should be k *= 2.
As mentioned before, it would be very good to move all this logic to using identity1D and zeros1D to avoid this sort of bugs...
Can you also add a test that exercises this path?

Reply:
Yes, as @mobicham mentioned, it doesn't work with `BLOCK_K > 128` currently.
- I'll fix this issue,
- I'll modify these logics to use `identity1D` and `zeros1D` as much as possible.

Thanks!